pretyping/patternops.ml

```diff
@@ -430,8 +430,16 @@ let rec pat_of_raw metas vars = DAst.with_loc_val (fun ?loc -> function
       with Glob_ops.ComplexSort -> user_err ?loc (str "Unexpected universe in pattern."))
   | GHole _ ->
       PMeta None
-  | GCast (c,_,_) ->
-      warn_cast_in_pattern ();
+  | GCast (c,_,t) ->
+      let () =
+        let ometas = !metas in
+        let _ : constr_pattern = pat_of_raw metas vars t in
+        if ometas != !metas then
```

Comment:
Physical equality on a type that is not enforced locally is asking for trouble when the code changes at distance. Even if slightly less efficient this should be turned into an equality check. (You can just check for the length, actually.)

Reply:
Pushed a version that doesn't rely on equality.